edge/pkg/edgehub/task/taskexecutor/pre_check.go

```diff
@@ -0,0 +1,142 @@
+package taskexecutor
+
+import (
+	"encoding/json"
+	"fmt"
+	"time"
+
+	"github.com/shirou/gopsutil/cpu"
+	"github.com/shirou/gopsutil/disk"
+	"github.com/shirou/gopsutil/v3/mem"
+
+	"github.com/kubeedge/kubeedge/common/types"
+	api "github.com/kubeedge/kubeedge/pkg/apis/fsm/v1alpha1"
+	"github.com/kubeedge/kubeedge/pkg/util/fsm"
+)
+
+const (
+	MaxCPUUsage  float64 = 80
+	MaxMemUsage  float64 = 80
+	MaxDiskUsage float64 = 80
+)
+
+func preCheck(taskReq types.NodeTaskRequest) fsm.Event {
```

Comment:
Will these functions be used for the image prepull?

Reply:
It will.